docs/adr/adr-013-tail-padding-optimizations.md

```diff
@@ -0,0 +1,55 @@
+# ADR 13: Tail Padding Optimizations
+
+## Changelog
+
+- 24.02.2023: First Draft
+
+## Author
+
+@nashqueue
+
+## Context
+
+If the Celestia block is not full it will create tail padding. Tail padded shares look always the same. Tail padded shares create tail padded rows. tail padded rows erasure to erasure-coded tail padded rows. Tail padded rows combined with erasure-coded tail padded rows result in a full tail padded row. A full tail padded row of the same length will always have the same Namespace Merkle Trie on top of it. This will result in a tail padded row root.
+All of those things are deterministic and can be computed beforehand.
```

Comment:
[no change needed][informational] FYI to celestia-node implementooors, the tail padding share schema has changed slightly since initial implementation but the [specs](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#tail-padding-share) match the current implementation. Additionally celestia-app exposes [`TailPaddingShares`](https://pkg.go.dev/github.com/celestiaorg/celestia-app@v0.11.1/pkg/shares#TailPaddingShares) which can be used to generate a tail padding share. 

celestia-app can consider exporting a constant for the row root for an extended data square tail padding row if celestia-node doesn't want to compute this itself

Reply:
I should add this to reference.